internal/reader/onchain_prices_reader.go

```diff
@@ -54,7 +55,14 @@ func (pr *OnchainTokenPricesReader) GetTokenPricesUSD(
 			if staticPrice, exists := pr.TokenPriceConfig.StaticPrices[token]; exists {
 				price.Set(&staticPrice)
 			} else {
-				if err := pr.ContractReader.GetLatestValue(ctx, contractName, functionName, token, price); err != nil {
+				if err :=
```

Comment:
style nit: I'm personally not a fan of "one liner" if conditions when they span multiple lines.